nixos/modules/services/web-apps/alist.nix

```diff
@@ -0,0 +1,313 @@
+                  lib.types.port
+                ];
+                default = (-1);
+                description = ''
+                  The https port to listen on, default `-1`, if you want to enable https, set it to non `-1`
+                '';
+              };
+              force_https = lib.mkOption {
+                type = lib.types.bool;
+                default = false;
+                description = ''
+                  Whether HTTPS protocol is forced, if true the user can only access the website through HTTPS.
+                '';
+              };
+              cert_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "Path of cert file";
+              };
+              key_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "Path of key file";
+              };
+              unix_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  Unix socket file path to listen on, default empty, if you want to use unix socket, set it to non empty
+                '';
+              };
+              unix_file_perm = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  Unix socket file permission, set to the appropriate permissions
+                '';
+              };
+            };
+            temp_dir = lib.mkOption {
+              type = lib.types.str;
+              default = "/var/lib/alist/temp";
+              description = ''
+                The directory to keep temporary files.
+                temp_dir is a temporary folder exclusive to alist. In order to prevent AList from generating garbage files when being interrupted, the directory will be cleared every time AList starts, so do not store anything in this directory.
+              '';
+            };
+          };
+        };
+        default = { };
+        description = ''
+          The alist configuration, see https://alist.nn.ci/config/configuration.html for documentation.
+
+          Options containing secret data should be set to an attribute set containing the attribute `_secret` - a string pointing to a file containing the value the option should be set to.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.alist = lib.optionalAttrs (cfg.user == "alist") {
+      description = "Alist user";
+      isSystemUser = true;
+      group = cfg.group;
+    };
+    users.groups.alist = lib.optionalAttrs (cfg.group == "alist") { };
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.alist = {
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart =
+        ''
+          umask 0077
+          mkdir -p ${cfg.dataDir}
```

Comment:
`umask` can be set in the systemd configuration.

The `mkdir` is not required if using `workingDirectory` directive.

Reply:
See the open-webui service for some inspiration.